metricbeat/module/system/socket_summary/socket_summary.go

```diff
@@ -0,0 +1,87 @@
+package socket_summary
+
+import (
+	"syscall"
+
+	"github.com/elastic/beats/libbeat/common"
+
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/metricbeat/mb"
+
+	"github.com/shirou/gopsutil/net"
+)
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each host defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet("system", "socket_summary", New)
```

Comment:
A new feature is that the namespace can now also be set during registration time: https://github.com/elastic/beats/blob/master/metricbeat/module/logstash/node_stats/node_stats.go#L22